### PR TITLE
SSL verification enablement for mscolab

### PR DESCRIPTION
**Pull request by nilupulmanodya**
**Purpose of PR?**:

Fixes #
Implementation for SSL verification enablement of mscolab for developers

**Does this PR introduce a breaking change?**
No

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
Should need to merge #1974  before merging this

**Does this PR results in some Documentation changes?**
_If yes, include the list of Documentation changes_

**Checklist:**
- [ ] Bug fix. Fixes #<issue number>
- [ ] New feature (Non-API breaking changes that adds functionality)
- [ ] PR Title follows the convention of  `<type>: <subject>`
- [ ] Commit has unit tests

### CODE REVIEW

**Review by matrss — APPROVED** [review]
LGTM if we need this.

For what it's worth I see one alternative implementation, where we wouldn't need to change anything under mslib/:
A `requests.Session` object has a `verify` attribute which is used as a default value in case nothing is specified. This attribute is set to true in `Session`s `__init__` (https://github.com/psf/requests/blob/f5a7aefc2dc123c65ec1c6d3e83949999b03876b/src/requests/sessions.py#L424).
All requests, even those that use the `requests.{get,post,...}` functions, do use such a Session object. We could therefore monkeypatch this class in our test suite (probably in conftest.py) to always set the verify attribute to false.

I am indifferent to how we do it. Your approach is more verbose, while what I described above would be a bit "hacky" / less obvious / harder to understand if we ever run into some issue with this.

**Comment by nilupulmanodya** [comment]
based on the discussion https://mss-devel.slack.com/archives/C04VCNH81MJ/p1696760025937339
The implementation is a kind of misunderstanding. Will do SSL verification for SAML later after #2043 is merged.

**Pull request closed without merging**
